weave

```diff
@@ -145,9 +177,11 @@ collect_ip_args() {
 if [ "$1" = "run" ] ; then
     shift 1
     check_docker_version
-    if [ "$1" = "--with-dns" ] ; then
-        shift 1
-        DNS_ARGS=$(exec_remote dns-args "$@")
+    collect_dns_args "$@"
```

Comment:
We did it for parity with the proxy implementation - that also errors when you specify both options.